pkgs/applications/virtualization/spim/default.nix

```diff
@@ -0,0 +1,81 @@
+{ mkDerivation, lib, fetchsvn, qmake, qtbase, qttools, bison, flex }:
+
+mkDerivation rec {
+  pname = "QtSpim";
+  version = "722";
+
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/spimsimulator/code";
+    rev = version;
+    sha256 = "1hfz41ra93pdd2pri5hibl63sg9yyk12a8nhdkmgj7h9bwgqxw6b";
+  };
+
+  sourceRoot = "code-r${version}/QtSpim";
+
+  nativeBuildInputs = [ bison flex qmake ];
+
+  buildInputs = [ qtbase qttools ];
+
+  # Remove build artifacts from the repo
+  preConfigure = ''
+    rm parser_yacc.h
+    rm parser_yacc.cpp
+    rm scanner_lex.cpp
+
+    rm help/qtspim.qhc
+  '';
+
+  # Fix bug in a generated Makefile
+  postConfigure = ''
+    substituteInPlace Makefile --replace '$(MOVE) help/qtspim.qhc help/qtspim.qhc;' ""
+  '';
+
+  # Fix documentation path
+  postPatch = ''
+    substituteInPlace menu.cpp --replace "/usr/lib/qtspim/help/qtspim.qhc" "$out/share/qtspim/help/qtspim.qhc"
+    substituteInPlace menu.cpp --replace "/usr/lib/qtspim/bin/assistant" "${qttools.dev}/bin/assistant"
+
+    substituteInPlace ../Setup/qtspim_debian_deployment/qtspim.desktop \
+      --replace "/usr/lib/qtspim/qtspim.png" "$out/share/qtspim/qtspim.png" \
+      --replace "/usr/bin/qtspim" "$out/bin/qtspim"
+  '';
+
+  buildPhase = ''
+    export QT_PLUGIN_PATH="${qtbase.bin}/${qtbase.qtPluginPrefix}"
+    export QT_QPA_PLATFORM_PLUGIN_PATH="${qtbase.bin}/${qtbase.qtPluginPrefix}/platforms"
+    make
```

Comment:
@ttuegel is there a better way to make `qhelpgenerator` inside build work than setting these vars?

Reply:
(about 'Outdated) moved it to `preBuild`, question is still valid

Reply:
(moved it to the `buildPhase` again, as `preBuild` version sometimes broke down when building with more than 3 `--cores`)

Reply:
I'm not sure why setting `QT_PLUGIN_PATH` isn't sufficient, and it should be OK to set these in `preBuild`, but I don't see anything wrong with this.